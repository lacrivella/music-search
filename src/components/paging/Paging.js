import React from 'react';
import PropTypes from 'prop-types';

function Paging({ onPrevious, onNext, currentPage, totalPages }) {
  return (
    <span>
      <button onClick={onPrevious}>↲</button>
      <p>Page {currentPage} of {totalPages}</p>
      <button onClick={onNext}>↳</button>
    </span>
  );
}

Paging.propTypes = {
  onPrevious: PropTypes.func.isRequired,
  onNext: PropTypes.func.isRequired,
  currentPage: PropTypes.string.isRequired,
  totalPages: PropTypes.string.isRequired,

}

export default Paging;
