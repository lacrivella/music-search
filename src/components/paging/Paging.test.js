import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(<Paging onPrevious={() => {}} onNext={() => {}} currentPage={6} totalPages={80}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
