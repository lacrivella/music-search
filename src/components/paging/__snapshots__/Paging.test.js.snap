// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<span>
  <button
    onClick={[Function]}
  >
    ↲
  </button>
  <span>
    Page 
    6
     of 
    80
  </span>
  <button
    onClick={[Function]}
  >
    ↳
  </button>
</span>
`;
